spec/spellchecker-spec.coffee

```diff
@@ -11,6 +11,54 @@ describe "SpellChecker", ->
     it "throws an exception when no word specified", ->
       expect(-> SpellChecker.isMisspelled()).toThrow()
 
+  describe ".checkSpelling(string)", ->
+    it "returns an array of character ranges of misspelled words", ->
+      string = "cat caat dog dooog"
+
+      expect(SpellChecker.checkSpelling(string)).toEqual [
+        {start: 4, end: 8},
+        {start: 13, end: 18},
+      ]
+
+    it "accounts for UTF16 pairs", ->
+      string = "😎 cat caat dog dooog"
```

Comment:
For Mac and Windows, this didn't require any extra work, because the `NSRanges` returned by `NSSpellChecker` (and probably all `NSString` APIs) seem to refer to UTF16 code point indices, as opposed to logical character indices, and the same applies for the windows APIs.

For Linux, the Hunspell library only provides a per-word spell-checking API; it doesn't handle arbitrary text. It also expects UTF8-encoded words. I deal with this by passing the string to the native spell-checkers in UTF16 (as V8 natively stores it), and for hunspell, transcoding to UTF8 one word at a time.